notification/store.go

```diff
@@ -297,6 +313,28 @@ func (db *DB) VerifyContactMethod(ctx context.Context, cmID string, code int) er
 	return nil
 }
 
+func messageStateFromStatus(lastStatus string, hasNextRetry bool) MessageState {
+	switch lastStatus {
+	case "queued_remotely", "sending":
+		return MessageStateSending
+	case "pending":
+		return MessageStatePending
+	case "sent":
+		return MessageStateSent
+	case "delivered":
+		return MessageStateDelivered
+	case "failed", "bundled": // bundled message was not sent (replaced) and should never be re-sent
+		// temporary if retry
+		if hasNextRetry {
+			return MessageStateFailedTemp
+		} else {
+			return MessageStateFailedPerm
+		}
+	default:
+		return -1
```

Comment:
It looks like both places we use this we are checking for `-1` and converting it to an error. Instead, let's just have this func return an error as a second argument so we can be explicit and return the same error for both.